tests/EDD/test_calibration.py

```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+from typing import Union
+import numpy as np
+import h5py
+from easistrain.EDD.calibrationEDD import calibEdd
+
+
+def generate_config(tmp_path: Path) -> dict:
+    HERE = Path(__file__).parent.resolve()
+    return {
+        "fileRead": str(tmp_path / "input_file.h5"),
+        "fileSave": str(tmp_path / "output_file.h5"),
+        "sample": "sample",
+        "dataset": "0000",
+        "scanNumberHorizontalDetector": 2,
+        "scanNumberVerticalDetector": 1,
+        "nameHorizontalDetector": "horz_detector",
+        "nameVerticalDetector": "vert_detector",
+        "numberOfBoxes": 4,
+        "nbPeaksInBoxes": [1, 2, 1, 1],
+        "rangeFit": [620, 780, 1020, 1120, 3500, 3800, 3850, 4090],
+        "sourceCalibrantFile": str(HERE.parent.parent / "Calibrants" / "BaSource"),
+    }
+
+
+def generate_input_file(cfg: dict, test_data_path: Union[Path, str]):
+    sample, dataset = cfg["sample"], cfg["dataset"]
+    n_scan_h, name_h = (
+        cfg["scanNumberHorizontalDetector"],
+        cfg["nameHorizontalDetector"],
+    )
+    n_scan_v, name_v = (
+        cfg["scanNumberVerticalDetector"],
+        cfg["nameVerticalDetector"],
+    )
+    with h5py.File(test_data_path, "r") as testFile:
+        with h5py.File(cfg["fileRead"], "w") as h5file:
+            h5file[f"{sample}_{dataset}_{n_scan_h}.1/measurement/{name_h}"] = testFile[
+                "horizontal/data"
+            ][()]
+            h5file[f"{sample}_{dataset}_{n_scan_v}.1/measurement/{name_v}"] = testFile[
+                "vertical/data"
+            ][()]
+
+
+def test_calibEdd(tmp_path: Path):
+    # Arrange
+    config = generate_config(tmp_path)
+    test_data_path = (
+        Path(__file__).parent.parent.resolve() / "data" / "Ba_calibration_data.hdf5"
+    )
+    with h5py.File(test_data_path, "r") as h5file:
+        ref_vertical_coeffs = h5file["vertical/coeffs"][()]
+        ref_horizontal_coeffs = h5file["horizontal/coeffs"][()]
+    generate_input_file(config, test_data_path)
+
+    # Act
+    calibEdd(**config)
+
+    # Assert
+    with h5py.File(config["fileSave"], "r") as h5file:
+        grp_name = f'fit_{config["dataset"]}_{config["scanNumberHorizontalDetector"]}_{config["scanNumberVerticalDetector"]}'
+        computed_vertical_coeffs = h5file[
+            f"detectorCalibration/{grp_name}/calibCoeffs/calibCoeffsVD"
+        ][()]
+        computed_horizontal_coeffs = h5file[
+            f"detectorCalibration/{grp_name}/calibCoeffs/calibCoeffsHD"
+        ][()]
+
+    np.testing.assert_allclose(computed_vertical_coeffs, ref_vertical_coeffs)
+    np.testing.assert_allclose(computed_horizontal_coeffs, ref_horizontal_coeffs)
```

Comment:
Then, if `calibEdd` ran fine, I extract the computed coeffs and compare them to the reference coeffs using [np.testing.assert_allclose](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy-testing-assert-allclose). If those are too different (`rtol > 1e-7`), the test fails.

Reply:
Could we set the difference as a % or maybe better if we set it to >= to the uncertainty of the correct parameters.

Reply:
> if we set it to >= to the uncertainty of the correct parameters.

Sounds good, I will change this !